### PR TITLE
dex sector subproject creation

### PR DESCRIPTION
**Pull request by jeff-dude**
update:
- `dbt compile` is complete on main spellbook project ✅ 
- `dbt compile` is complete on dex project ✅
- ~~run subproject end-to-end (increase CI cluster if needed)~~
  - this may not be feasible and/or necessary. similar to nft PR, we can likely just merge with modified turned off.
- cleanup dbt project file on main spellbook ✅ 
- ~~...maybe more~~

### CODE REVIEW

**Comment by jeff-dude** [comment]
main spellbook project ran all models successfully:
https://github.com/duneanalytics/spellbook/actions/runs/9519362533/job/26242396617

an old test failed, but we don't run in prod or ongoing basis, so not worried about it. considering main spellbook project fine in this PR.

**Comment by jeff-dude** [comment]
attempting to run the full dex subproject with a dummy change to CI pipeline to remove `state:modified` so all runs:
https://github.com/duneanalytics/spellbook/pull/6161/commits/1672016368ec3ce66226ab8b6e0f70b4786a7ad9
https://github.com/duneanalytics/spellbook/actions/runs/9520100813/job/26244663920?pr=6161

File: dex/tests/cow_protocol/ethereum/cow_protocol_ethereum_assert_partial_fill.sql [hunk 1]
```diff
@@ -1,37 +0,0 @@
--- Given a list of trades, when we look at their partial_fill, based on the type of the trade we expect partial_fill to behave differently
```
Comment (aalan3) [inline dex/tests/cow_protocol/ethereum/cow_protocol_ethereum_assert_partial_fill.sql]: Don't we want to keep these tests?

Comment (jeff-dude) [reply dex/tests/cow_protocol/ethereum/cow_protocol_ethereum_assert_partial_fill.sql]: they were valuable in development phase, but don't really add value moving forward. we don't run the tests any time and it's hyper focused on one tx. we can keep, but feels like a good opportunity to cleanse the project

**Comment by jeff-dude** [comment]
i'm considering this ready to go. feel free to give final review monday morning and push any changes you see fit. happy to help finalize when i'm online monday, otherwise you can do before i'm online 🤝

**Review by aalan3 — APPROVED** [review]

**Pull request merged**